broker/unified_sql/src/stream_sql.cc

```diff
@@ -2650,7 +2652,7 @@ void stream::_check_and_update_index_cache(const Service& ss) {
     _index_data_insert.bind_value_as_i32(2, ss.service_id());
     _index_data_insert.bind_value_as_str(3, sv);
     _index_data_insert.bind_value_as_str(4, "0");
-    _index_data_insert.bind_value_as_bool(5, special);
+    _index_data_insert.bind_value_as_bool(5, special ? "1" : "0");
```

Comment:
this method has the following prototype:
void mysql_stmt::bind_value_as_bool(int range, bool value);
As you can see the second argument is a boolean and not a char*.
In your case, this is cast to boolean and as "1" or "0" is not a nullptr, they are both considered as true.

Reply:
So I think you introduced a bug here.